Serializer/Hateoas/SonataRelationProviderResolver.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/*
+ * This file is part of the Sonata package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ */
+namespace Sonata\AdminBundle\Serializer\Hateoas;
+
+use Hateoas\Configuration\Provider\Resolver\RelationProviderResolverInterface;
+use Sonata\AdminBundle\Admin\Pool;
+use Hateoas\Configuration\RelationProvider;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Hateoas\Configuration\Relation;
+use Hateoas\Configuration\Route;
+
+/**
+ * @author Daniel Leech <daniel@dantleech.com>
+ */
+class SonataRelationProviderResolver implements RelationProviderResolverInterface
+{
+    /**
+     * @var Pool
+     */
+    private $pool;
+
+    /**
+     * @var UrlGeneratorInterface
+     */
+    private $urlGenerator;
+
+    /**
+     * @param Pool $pool
+     * @param UrlGeneratorInterface $urlGenerator
+     */
+    public function __construct(Pool $pool, UrlGeneratorInterface $urlGenerator)
+    {
+        $this->pool = $pool;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getRelationProvider(RelationProvider $relationProvider, $object)
+    {
+        if (false === $this->pool->hasAdminByClass(get_class($object))) {
+            return null;
+        }
+
+        return array($this, 'getRelations');
+    }
+
+    /**
+     * Return the sonata admin relations
+     *
+     * @return array
+     */
+    public function getRelations($object)
```

Comment:
Hmm. Actually my second point doesn't work.

For the tree admin browser the link would have to have the name for each document otherwise we wouldn't know how to pick the correct one.

I guess we would have to "break" hateoas and have something like

````
{
    '_links': {
        'edit': {
            'admin1': 'path/to',
            'admin2': 'path/to',
        }
    }
}
````

But then we also have the problem of how to refer to these different admins in the menu.

/cc @WouterJ